### ISSUE TITLE
prefer-explicit-assert rule allows to avoid assertions if wrapped on expect() or using toThrow()

### ISSUE THREAD

**Issue by Miguelerja** [testing-library/eslint-plugin-testing-library#159]
Hello,

I have found two cases that I think should be controlled by this rule but are not being detected by linter.

If wrapped on an expect with no assertion it will not throw a warning/error although it is not explicit either.
<img width="688" alt="Screenshot 2020-06-11 at 08 54 41" src="https://user-images.githubusercontent.com/42043025/84355116-5b2e7d80-abc2-11ea-8595-463dd5572104.png">

If using jest's toThrow assertion it doesn't throw a warning/error although, even though it is an explicit assertion, it is relying on the same case of throwing an error to test the application.
<img width="522" alt="Screenshot 2020-06-11 at 08 53 09" src="https://user-images.githubusercontent.com/42043025/84355185-74cfc500-abc2-11ea-9068-41533a1eba54.png">

**Comment by Belco90** [comment]
Hey Miguel. I'm not really sure if this should be a concern for the rule (or even the plugin itself).

We have had similar discussions before around how far the plugin should check stuff regarding jest or any other testing framework.

Anyway, happy to see what is your proposal for this. Keep throwing an error until the rule is wrapped with `expect` + some matcher?

@gndelia @timdeschryver thoughts?

**Comment by Miguelerja** [comment]
I see your point. On the other hand I think that, if the rule's reason to be is avoiding assertions based on a not found error but make people write defined and expressive assertions, these cases somehow go against this philosophy.

I think the first case is equal to just say ```getByRole(<role>)``` and is a kind of hack over the rule since just expect with no expectation is not adding any real meaning. 

On the second case though, I agree it can go beyond the rule scope since it is indeed a full assertion.

**Comment by Belco90** [comment]
I get what you mean, but I don't know if we would be crossing boundaries from testing library. [`eslint-plugin-jest` has already a rule to check valid expects](https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/valid-expect.md) so this plugin would be doing kind of the same?

Maybe we can get some inspiration from that rule, but in a simpler way. I still think it's out of the scope of this plugin, but I'll re-think about it.

**Comment by timdeschryver** [comment]
I was also thinking that way @Belco90 .
This is something that's not in the "domain" of Testing Library.

About the second case (with `.ToThrow()`), I think this is valid, but I'm not sure... I think it could be used to verify that an element isn't on the page?

**Comment by gndelia** [comment]
By reading the docs of [prefer-explicit-assert](https://github.com/testing-library/eslint-plugin-testing-library/blob/master/docs/rules/prefer-explicit-assert.md) I feel the first scenario `expect(getByText('foo'))` should be covered

not so sure about the second one yet.  I feel the second scenario is similar to https://github.com/testing-library/eslint-plugin-testing-library/issues/110 where we are mixing the RTL queries/methods with matches that come from jest.
